server/server/src/file_service.rs

```diff
@@ -330,195 +348,188 @@ pub async fn get_document(
-                        "attempted to disappear a file, the owner was not present, id: {}, owner: {:?}",
-                        id,
-                        owner
-                    );
                 }
+            }
+        }
 
-                // maintain index: shared_files
-                for user_access_key in meta.user_access_keys() {
-                    let sharee = Owner(user_access_key.encrypted_for);
-                    if let Some(mut shared_files) = tx.shared_files.delete(sharee) {
-                        if !shared_files.remove(&id) {
-                            error!(?id, ?sharee, "attempted to disappear a file, a sharee didn't have it shared");
-                        }
-                        tx.shared_files.insert(sharee, shared_files);
-                    } else {
-                        error!(
-                            "attempted to disappear a file, the sharee was not present, id: {}, sharee: {:?}",
-                            id,
-                            sharee
-                        );
-                    }
-                }
+        for id in metas_to_delete {
+            let meta = db
+                .metas
+                .remove(&id)?
+                .ok_or(ClientError(AdminDisappearFileError::FileNonexistent))?;
+
+            // maintain index: owned_files
+            let owner = meta.owner();
+
+            if !db.owned_files.remove(&owner, &id)? {
+                error!(
+                    ?id,
+                    ?owner,
+                    "attempted to disappear a file, owner or id not present in owned_files"
+                );
+            }
 
-                // maintain index: file_children
-                let parent = *meta.parent();
-                if let Some(mut file_children) = tx.file_children.delete(*meta.parent()) {
-                    if !file_children.remove(&id) {
-                        error!(?id, ?parent, "attempted to disappear a file, the parent didn't have it as a child");
-                    }
-                    tx.file_children.insert(*meta.parent(), file_children);
-                } else {
+            // maintain index: shared_files
+            for user_access_key in meta.user_access_keys() {
+                let sharee = Owner(user_access_key.encrypted_for);
+                if !db.shared_files.remove(&sharee, &id)? {
                     error!(
-                        "attempted to disappear a file, the parent was not present, id: {}, parent: {:?}",
-                        id,
-                        meta.parent()
+                        ?id,
+                        ?sharee,
+                        "attempted to disappear a file, a sharee didn't have it shared"
                     );
                 }
             }
 
-            Ok(docs_to_delete)
-        })?;
+            // maintain index: file_children
+            let parent = *meta.parent();
+            if !db.file_children.remove(meta.parent(), &id)? {
```

Comment:
this doesn't seem to match the previous implementation, which also removed the set for `meta.parent()` entirely